build-support/bin/ci.sh

```diff
@@ -135,9 +135,8 @@ fi
 
 # NB: Ordering matters here. We (currently) always bootstrap a Python 2 pex.
 if [[ "${python_three:-false}" == "true" ]]; then
-  # The 3.4 end of this constraint is necessary to jive with the travis ubuntu trusty image.
   banner "Setting interpreter constraints for 3!"
-  export PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS='["CPython>=3.4,<4"]'
+  export PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS='["CPython>=3.5,<4"]'
```

Comment:
This was fixed by https://github.com/pantsbuild/pants/commit/7da3ab500613f2b33633f3180b7460adfcdc5e35, but we didn't notice then. 

I think this might be fine for scope creep, because it directly ties to the failing test, but let me know.